### PR TITLE
pkgs/by-name/[abc].+ Migrate to new Darwin SDK

### PR DESCRIPTION
**Pull request by sarahec**
1. Removed references to darwin.apple_sdk.frameworks.
2. Automatically formatted to RFC style.

Tracking: #354146

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 355564` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>alcom</li>
    <li>alexandria</li>
    <li>insulator2</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>gitbutler</li>
  </ul>
</details>
<details>
  <summary>39 packages built:</summary>
  <ul>
    <li>aegisub</li>
    <li>aiken</li>
    <li>amp</li>
    <li>arti</li>
    <li>attic-client</li>
    <li>attic-server</li>
    <li>azurite</li>
    <li>binsider</li>
    <li>browsers</li>
    <li>bunbun</li>
    <li>c2patool</li>
    <li>caligula</li>
    <li>cargo-bump</li>
    <li>cargo-geiger</li>
    <li>cargo-information</li>
    <li>cargo-make</li>
    <li>cargo-modules</li>
    <li>cargo-raze</li>
    <li>cargo-tauri</li>
    <li>cargo-unfmt</li>
    <li>cargo-vibe</li>
    <li>cargo-wasi</li>
    <li>cargo-xwin</li>
    <li>citations</li>
    <li>clipcat</li>
    <li>code2prompt</li>
    <li>codeberg-cli</li>
    <li>committed</li>
    <li>convco</li>
    <li>cook-cli</li>
    <li>crabfit-api</li>
    <li>crunchy-cli</li>
    <li>csvlens</li>
    <li>czkawka</li>
    <li>czkawka-full</li>
    <li>en-croissant</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
    <li>open-scq30</li>
  </ul>
</details>

File: pkgs/by-name/ae/aegisub/package.nix [hunk 1]
```diff
@@ -78,43 +68,36 @@ stdenv.mkDerivation (finalAttrs: {
...
+      libiconv
+      libuchardet
+      pcre
+      wxGTK
+      zlib
+    ]
+    ++ lib.optionals alsaSupport [ alsa-lib ]
+    ++ lib.optionals openalSupport [
+      (if stdenv.hostPlatform.isDarwin then darwin.apple_sdk.frameworks.OpenAL else openal)
+    ]
```
Comment (tjni) [inline pkgs/by-name/ae/aegisub/package.nix:97]: This is actually kind of subtle (and you might want to extract this into its own PR for clarity):

After the refactor, on Darwin, OpenAL is always present, so this part can simply be:

```
++ lib.optionals (openalSupport && !stdenv.hostPlatform.isDarwin) [
  openal
]
```

But in order to properly disable openal on Darwin, we now need

```
cmakeFlags = [ "-DWITH_OPENAI=OFF" ];
```

when `openalSupport` is false.

Comment (sarahec) [reply pkgs/by-name/ae/aegisub/package.nix:97]: I like the idea of dropping this in its own PR for clarity. That way we can document the change properly.

Comment (sarahec) [reply pkgs/by-name/ae/aegisub/package.nix]: I had to move the patch into here due to conflicts with the new default SDK

**Comment by sarahec** [comment]
Closing, as we have the proof the new SDK structure migrates easily.
